cookbook/bundles/tests.rst

```diff
@@ -0,0 +1,311 @@
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+In this example, Symfony is used as an application that mimicks the HTTP protocol,
+which means that it is possible to write transversal tests (System Tests) by building
+"manually" a ``Request``, giving it to our application and then check the returned ``Response``.
+
+This article provides examples on how to write such tests with third party bundles.
+
+Creating a Minimal AppKernel
+----------------------------
+
+Third party bundles cannot be ran standalone, they need an application.
+It is possible to create a minimalistic ``AppKernel`` inside the bundle for testing
+and showcase purposes, instead of:
+
+# Creating a separate application;
+# Install the bundle in it;
+# Then check how things turn out.
+
+If the third party bundle is named ``AcmeMyBundle``, the ``AppKernel`` could look like this::
+
+    // Tests/app/AppKernel.php
+    namespace Acme\MyBundle\Tests;
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            return array(
+                new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+                new Acme\MyBundle\AcmeMyBundle(),
+            );
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load(__DIR__.'/config.yml');
+        }
+    }
+
+.. note::
+
+    ``AppKernel`` adds a direct dependency on Symfony's FrameworkBundle, which can be
+    installed by running the following command: ``composer require --dev symfony/framework-bundle``.
+
+In its minimum state ``AppKernel`` requires only one configuration parameter, ``secret``:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # Tests/app/config.yml
+        framework:
+            secret: Th1s1sS3cr3t!
+
+Symfony applications generate cache and logs directories, which can be ignored by
+adding those lines to ``.gitignore``:
+
+.. code-block::
+
+    # .gitignore
+    /Tests/app/cache
+    /Tests/app/logs
+    /vendor
+    /composer.lock
+
+Browsing Pages
+--------------
+
+If the third party bundle provides a controller, it can be helpful to create a
+front controller similar to the one that can be found in Symfony's Standard Edition::
+
+    // Tests/app/web.php
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/autoload.php';
+
+    $kernel = new AppKernel('prod', false);
+    $request = Request::createFromGlobals();
+    $response = $kernel->handle($request);
+    $response->send();
+
+With this it becomes possible to browse the page. A way to do it without having to configure
+a web server is to run the following command:
+
+.. code-block:: bash
+
+    $ php Tests/app/console.php server:run -d Tests/app
```

Comment:
I propose to convert this to a simple sidebar and adding it after the Running Commands section. Like:

```rst
.. sidebar:: Visiting Pages in the Browser

    As you can run commands now, it is possible to run the internal webserver
    to visit the pages in a browser. In order to do this, you have to add a minimal
    front controller (just like the one that can be found in the Symfony Standard
    Edition::

        // Tests/Functional/app/web.php
        use Symfony\Component\HttpFoundation\Request;

        require_once __DIR__.'/autoload.php';

        $kernel = new AppKernel('prod', false);
        $request = Request::createFromGlobals();
        $response = $kernel->handle($request);
        $response->send();

    Now, you can execute the ``server:run`` comand:

    .. code-block:: bash

        $ php Tests/Functional/app/console.php server:run -d Tests/app
```